### ISSUE TITLE
Calendar's navigator template onChange param not working

### ISSUE THREAD

**Issue by glencoden** [primefaces/primereact#2646]
### There is no guarantee in receiving an immediate response in GitHub Issue Tracker, If you'd like to secure our response, you may consider *PrimeReact PRO Support* where support is provided within 4 business hours

**I'm submitting a ...**  (check one with "x")
```
[ x] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

**Codesandbox Case (Bug Reports)**
Please fork the codesandbox below and create a case demonstrating your bug report. Issues without a codesandbox have much less possibility to be reviewed.

**Current behavior**

I provide prime react's DropDown as a navigator template for Calendar. In the DropDown's onChange handler, I call the onChange passed in the params to my template.

`const navigatorTemplate: React.FC<
    CalendarMonthNavigatorTemplateParams | CalendarYearNavigatorTemplateParams
> = ({ options, value, onChange }) => {
    return (
        <Dropdown
            className={style.NavigatorTemplate}
            options={options}
            value={value}
            onChange={({ originalEvent, value: changeValue }: DropdownChangeParams) => {
                onChange({ event: originalEvent, value: changeValue })
            }}
        />
    )
}`

**Expected behavior**

I expect this to change the Calendar's state, when a dropdown value is selected, but it doesn't.

This following work around is fine, although a bit hacky and hard to type:

`const navigatorTemplate: React.FC<
    CalendarMonthNavigatorTemplateParams | CalendarYearNavigatorTemplateParams
> = ({ options, value, element }) => {
    return (
        <Dropdown
            className={style.NavigatorTemplate}
            options={options}
            value={value}
            onChange={({ value }: DropdownChangeParams) => {
                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
                // @ts-ignore
                element?.props?.onChange({ target: { value } }) // work around because onChange({ value, event }) is not updating the Calendar
            }}
        />
    )
}`

**Minimal reproduction of the problem with instructions**

**Please tell us about your environment:**

VSC, npm, webpack

* **React version:**

17

* **PrimeReact version:**

7

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

Chrome
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
en

**Comment by melloware** [comment]
So i can't quite tell are you just saying its a TypeScript definition issue?

**Comment by glencoden** [comment]
No! When I pass, say, the yaer 2012 to the onChange handler passed as param to my yearHeaderTemplate, the Calendar will still be in 2022. Def a functionality problem.

**Comment by matheus-crivellari** [comment]
> So i can't quite tell are you just saying its a TypeScript definition issue?

This is probably a type definition issue as @melloware said. 

I just found that the `CalendarMonthNavigatorTemplateParams` states `onChange` must receive only one param of type `CalendarNavigatorTemplateChangeParams` but the actual calendar navigator `onChange` callback expects two params, the first is the `React.SyntheticEvent` and the second is the actual value.

I reached the desired behaviour by not defining the calendar navigator template callback params type and thus letting TypeScript infer the onChange signature, as below:

```typescript

function monthNavigatorTemplate({ value, options, className, onChange }) {
  return (
    <Dropdown
      value={value}
      options={options}
      className={className}
      onChange={(event) => {
        onChange(event.originalEvent, event.value);
      }}
    />
  );
}
```

Not the best approach I think, because the .d.ts files and the actual js are divergent. But it worked around.

**Comment by melloware** [comment]
Ok I will see if I can fix the typescript def and submit a PR.

**Comment by melloware** [comment]
Hmm but the `CalendarNavigatorTemplateChangeParams `is defined like this...

```ts
interface CalendarNavigatorTemplateChangeParams {
    event: React.SyntheticEvent;
    value: string | number | undefined | null;
}
```

Which is correct. So it looks liek this is defined correctly in the TS file?

**Comment by melloware** [comment]
Can you create a Code Sandbox reproducer so I can see the TS error you are getting? I have a feeling you just aren't setting something correctly.

Use: https://codesandbox.io/s/qjx332qq4

**Comment by glencoden** [comment]
Hello! I‘m on holiday but would love to help on 28.3. if it‘s soon enough.

Von meinem iPhone gesendet

**Comment by glencoden** [comment]
Hey,

I edited the sandbox project. Is it visible to you yet?

In the sandbox, the navigatorTemplate overwrite doesn't seem to work at all so far. Can you see why?

**Comment by glencoden** [comment]
Maybe here :) https://codesandbox.io/s/primereact-test-forked-qhd0nc?file=/src/index.js

**Comment by melloware** [comment]
yep i see it but this isn't TypeScript its vanilla JS?  So I don't understand above when you mention `ts-ignore` to fix it?

**Comment by glencoden** [comment]
Oh that's probably a left over of the copy/paste! Never mind it. The problem is not a typescript problem, it is the javascript functionality. I'll add some comments.

**Comment by melloware** [comment]
@glencoden any update on this?

**Comment by melloware** [comment]
Will close until i here back from OP.
